test/integration/src/test/java/io/pravega/test/integration/ReadTest.java

```diff
@@ -181,6 +181,10 @@ public void readThroughSegmentClient() throws SegmentSealedException, EndOfSegme
         SegmentInputStream in = segmentConsumerClient.createInputStreamForSegment(segment);
         ByteBuffer result = in.read();
         assertEquals(ByteBuffer.wrap(testString.getBytes()), result);
+
+        // Test large write followed by read
+        out.write(new PendingEvent(null, ByteBuffer.wrap(new byte[150000]), new CompletableFuture<>()));
+        in.read();
```

Comment:
This covers the case you found. But you should also add:
```
        out.write(new PendingEvent(null, ByteBuffer.wrap(new byte[15]), new CompletableFuture<>()));
        out.write(new PendingEvent(null, ByteBuffer.wrap(new byte[15]), new CompletableFuture<>()));
        out.write(new PendingEvent(null, ByteBuffer.wrap(new byte[150000]), new CompletableFuture<>()));
        in.read();
        in.read();
        in.read();
```
Which will fail the way the change is now.